hal/i2c_api.h

```diff
@@ -64,6 +64,14 @@ enum {
     I2C_ERROR_BUS_BUSY = -2
 };
 
+typedef struct {
+    const int peripheral;
```

Comment:
I'm not convinced all the members of the `pinmap_t` objects themselves should be const. Not sure that helps, and maybe could get in the way of something.

Reply:
I assumed that these fields must be const, but for one pin peripherals we use `PinMap` and its fields are not `const`:

```
typedef struct {
    PinName pin;
    int peripheral;
    int function;
} PinMap;
```

For consistency, I can drop const in multi-pin pin-map types.